### ISSUE TITLE
How to get .mat file for every image?

### ISSUE THREAD

**Issue by psuu0001** [zhangchenxu528/FACIAL#22]
Hi, I tried to make my own dataset. But I have no idea to create the mat file. When I use the deep3dface to make the mat file, it asked a txt files which record the coordinates of eye, noise and mouth. But the deep3dface author doesn't show how to make the txt file for every image. Could you please show us the method to make the mat file? or the method to make .txt file for every iamge?

**Comment by zhangchenxu528** [comment]
Thank you for your attention. The deep3dreconstruction method requires weakly labeled inputs. In fact, our Openface result can be used to provide this information (or use dlib method).

An example using openface to read 5 points:
  x0 = (csvinfo.iloc[c][' x_37'] + ​​csvinfo.iloc[c][' x_38'] + csvinfo.iloc[c][' x_41'] + csvinfo.iloc[c][' x_40'] )/4
  y0 = (csvinfo.iloc[c][' y_37'] + ​​csvinfo.iloc[c][' y_38'] + csvinfo.iloc[c][' y_41'] + csvinfo.iloc[c][' y_40'] )/4
  x1 = (csvinfo.iloc[c][' x_43'] + csvinfo.iloc[c][' x_44'] + csvinfo.iloc[c][' x_47'] + ​​csvinfo.iloc[c][' x_46'] )/4
  y1 = (csvinfo.iloc[c][' y_43'] + csvinfo.iloc[c][' y_44'] + csvinfo.iloc[c][' y_47'] + ​​csvinfo.iloc[c][' y_46'] )/4
  x2 = csvinfo.iloc[c][' x_30']
  y2 = csvinfo.iloc[c][' y_30']
  x3 = csvinfo.iloc[c][' x_48']
  y3 = csvinfo.iloc[c][' y_48']
  x4 = csvinfo.iloc[c][' x_54']
  y4 = csvinfo.iloc[c][' y_54']

**Comment by psuu0001** [comment]
Very appreciate for your reply. It really works.

**Comment by zhanghm1995** [comment]
@zhangchenxu528 , Hi, along with this question, I have a question that since we could also get the Pose information from Deep3DFace repo, if I get the five facial landmarks as the input to Deep3DFace and then get the Pose information. Is it means we could use the head pose directly?

Or you have some other considerations to get the head pose from OpenFace method?

**Comment by zhangchenxu528** [comment]
The pose information for Deep3Dface repo is based on cropped images, but we need to get the head pose for the whole image. So, Deep3Dface result can not be used directly.  For Openface pose, you need to adjust the coordinate for the face model, too.

3e also pide the fit_headpose.py method to fit the head pose for each image. This method is more convenient. Please see how to use it in the Colab demo.

**Comment by zhanghm1995** [comment]
@zhangchenxu528 , thanks for your quickly reply. I got your meanings by your descriptions. And I have noticed you just use the 68 facial landmarks and eyeblink AU information from OpenFace results, could I use some other facial landmarks detection algorithm to get these 68 facial landmarks since the OpenFace running environment is hard to configure if I ignore the eyeblink information?

BTW, I know in this work, you consider the implicit head pose movements from audio input, if I only consider the 3D mouth movements by using audio input, does it mean I don't need to fit the head pose to the original  face image? I only need to generate the 64-d expression parameters?
